src/riak_pb_ts_codec.erl

```diff
@@ -162,18 +160,21 @@ cell_for(undefined) ->
     #tscell{}.
 
 
--spec decode_row([#tsrow{}], list(tuple())) -> [tsrow()].
-decode_row([], Acc) ->
+-spec decode_rows([#tsrow{}], list(tuple())) -> [tsrow()].
+decode_rows([], Acc) ->
     lists:reverse(Acc);
-decode_row([#tsrow{cells = Row} | T], Acc) ->
-    decode_row(T, [list_to_tuple(decode_cells(Row, [])) | Acc]).
+decode_rows([#tsrow{cells = Row} | T], Acc) ->
+    decode_rows(T, [list_to_tuple(decode_cells(Row, [])) | Acc]).
 
 -spec decode_numeric(binary()) -> float().
 decode_numeric(Num) ->
     NumList = binary_to_list(Num),
     case string:chr(NumList, $.) of
         0 ->
-            list_to_float(string:concat(NumList, ".0"));
+            try list_to_integer(NumList)
+            catch error:badarg ->
+                    list_to_float(string:concat(NumList, ".0"))
```

Comment:
This was my attempt to have big integers (which are converted to numeric type on the way in, here https://github.com/basho/riak_pb/blob/fix/az/avoid-float_to_binary-in-deencoding-tscell/src/riak_pb_ts_codec.erl#L132 and, the way it was originally, here https://github.com/basho/riak_pb/blob/end-to-end/timeseries/src/riak_pb_ts_codec.erl#L138) restored back to integers when they get through the riak_pb_ts internals.